### ISSUE TITLE
Inpainting+ControlNet poor results with canny or hed

### ISSUE THREAD

**Issue by reimager** [huggingface/diffusers#3431]
### Describe the bug

I'm using the stable_diffusion_controlnet_inpaint pipeline - I know this is a community pipeline, but I'm filing here as it seems likely this will be added to the main src at some point.

Pretty much all the controlnet methods work as expected, except canny and hed produce weird results that I can't explain (example of depth producing expected results and canny producing unexpected results attached).

Anytime I use canny or hed, I get these weird washed out results. 

A couple data points:
- multi-controlnet involving canny or hed also produces weird results.
- changing the mask has no effect - I tried masking 100% of the photo which I expected to behave like regular controlnet pipeline, but the weird results still happen
- controlnet_img2img pipeline doesnt have this weirdness
- happens with all the SD1.5 models I have tried, so I don't think its model specific

Entirely possible I am doing something obviously wrong though, filing in case anyone has ideas or someone else encounters same issue.

![summary](https://github.com/huggingface/diffusers/assets/20918094/f6cee585-4919-48b4-b80c-f49121a8df0e)

### Reproduction

import torch
from PIL import Image
from pytorch_lightning import seed_everything
from diffusers.utils import load_image
from diffusers import (
    DiffusionPipeline,
    ControlNetModel,
)

image_input = load_image("http://metaloft.com/images/canny-input.png").convert("RGBA")
image_mask = load_image("http://metaloft.com/images/canny-mask.png").convert("1")
image_control = load_image("http://metaloft.com/images/canny-control.png").convert("RGB")

controlnet = ControlNetModel.from_pretrained("lllyasviel/control_v11p_sd15_canny", torch_dtype=torch.float16)
pipe = DiffusionPipeline.from_pretrained("runwayml/stable-diffusion-inpainting", custom_pipeline="stable_diffusion_controlnet_inpaint", torch_dtype=torch.float16, controlnet=controlnet, revision="fp16")
pipe.to("cuda")
pipe.enable_xformers_memory_efficient_attention()

seed_everything(0)

result = pipe(prompt=["a dog on a bench"],image=[image_input], mask_image=image_mask, width=512, height=512, guidance_scale=0.5, num_inference_steps=25, controlnet_conditioning_image=[image_control])
result.images[0].save(f"output.png")

### System Info

- `diffusers` version: 0.17.0.dev0
- Platform: Linux-5.10.0-22-amd64-x86_64-with-glibc2.29
- Python version: 3.8.10
- PyTorch version (GPU?): 2.1.0a0+fe05266 (True)
- Huggingface_hub version: 0.14.1
- Transformers version: 4.30.0.dev0
- Accelerate version: 0.20.0.dev0
- xFormers version: 0.0.20+a74599b.d20230514
- Using GPU in script?: yes, 3090ti
- Using distributed or parallel set-up in script?: no

**Comment by patrickvonplaten** [comment]
Hey @reimager, 

Yes I noticed this too, it's to a big part because the controlnets were not trained on the inpainting checkpoint. Reducing the number of steps and then contronet strength seems to help though! I've set the default to just 0.5 [here](https://github.com/huggingface/diffusers/pull/3386/files#r1195522198). 

Also this repo gives some nice explanations: https://github.com/mikonvergence/ControlNetInpaint

**Comment by reimager** [comment]
@patrickvonplaten 

I owe you a coffee! Thank you for the link - that perfectly explains whats going on, and yes I tested by reducing controlnet_conditioning_scale and it does help. Not sure why canny and hed require this and other don't but it clearly works around the issue.

**Comment by ghpkishore** [comment]
@patrickvonplaten can we use normal sd_v1.5 model in the inpainting pipeline? that way we can get the trained model to work with controlnets. Also the reason is because in A1111 SD webui, controlnet works with any base model. They aren't limited to inpainting models for inpainting pipeline.

**Comment by patrickvonplaten** [comment]
You can use https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_onnx_stable_diffusion_inpaint_legacy.py for this
